### PR TITLE
NFRR audit result with collectorItemId mapping

### PR DESCRIPTION
**Pull request by nameisaravind**
This PR is created to have collectorItemId mapping in audit result data collected by NFRR collector

### CODE REVIEW

File: api-audit/src/main/java/com/capitalone/dashboard/evaluator/RegressionTestResultEvaluator.java [hunk 1]
```diff
@@ -62,6 +62,10 @@
     private static final String FAILURE_COUNT = "failureCount";
     private static final String SKIP_COUNT = "skippedCount";
     private static final String TOTAL_COUNT = "totalCount";
```
Comment (Sbrenthughes) [inline api-audit/src/main/java/com/capitalone/dashboard/evaluator/RegressionTestResultEvaluator.java]: The audit api changes need to be in a separate PR

Comment (nameisaravind) [reply api-audit/src/main/java/com/capitalone/dashboard/evaluator/RegressionTestResultEvaluator.java]: It is moved to separate PR

File: collectors/audit/nfrr/src/main/java/com/capitalone/dashboard/collector/AuditCollectorTask.java [hunk 2]
```diff
@@ -32,33 +34,29 @@
...
     public AuditCollectorTask(TaskScheduler taskScheduler, DashboardRepository dashboardRepository,
                               AuditResultRepository auditResultRepository, AuditCollectorRepository auditCollectorRepository,
-                              AuditSettings settings) {
+                              CmdbRepository cmdbRepository, ComponentRepository componentRepository,
+                              CollectorItemRepository collectorItemRepository, AuditSettings settings) {
         super(taskScheduler, "AuditCollector");
         this.dashboardRepository = dashboardRepository;
         this.auditResultRepository = auditResultRepository;
         this.auditCollectorRepository = auditCollectorRepository;
+        this.cmdbRepository = cmdbRepository;
```
[the remark this answers is not in the document]
Comment (nameisaravind) [reply collectors/audit/nfrr/src/main/java/com/capitalone/dashboard/collector/AuditCollectorTask.java]: @Sbrenthughes To keep the @Autowire spring bean initializations restricted to one class(AuditCollectorTask.java) and to keep Util class free from bean initializations.

File: collectors/audit/nfrr/src/test/resources/collector_items/items.json [hunk 3]
```diff
@@ -502,5 +502,43 @@
...
+    "errors" : [
+
+    ],
+    "pushed" : true,
+    "collectorId" : "5bb70128039798589fd792ec",
+    "lastUpdated" : 1557101003714,
+    "options" : {
+      "options" : {
+        "jobName" : "test result audit job",
+        "instanceUrl" : "https://jira.company.com/browse/TR-21750"
```
Comment (Sbrenthughes) [inline collectors/audit/nfrr/src/test/resources/collector_items/items.json]: Can you update this to follow the collector Item for Audit Result

Comment (nameisaravind) [reply collectors/audit/nfrr/src/test/resources/collector_items/items.json]: @Sbrenthughes updated

**Comment by Sbrenthughes** [comment]
Can you update this branch? I will then approve and merge

**Comment by nameisaravind** [comment]
> Can you update this branch? I will then approve and merge

updated
